core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -102,7 +99,7 @@ public QueryBuilder parseTopLevelQueryBuilder() {
             String msg = "query malformed, empty clause found at [" + parser.getTokenLocation() +"]";
             DEPRECATION_LOGGER.deprecated(msg);
             if (parseFieldMatcher.isStrict()) {
-                throw new IllegalArgumentException(msg);
+                throw new ParsingException(parser.getTokenLocation(), "query malformed, empty clause found");
```

Comment:
maybe this should stay consistent with the exception that we throw whenever some deprecated syntax is used?